sdk/webpubsub/azure-webpubsub-client/README.md

```diff
@@ -0,0 +1,217 @@
+# Azure Web PubSub client library for Python
+
+[Azure Web PubSub](https://aka.ms/awps/doc) is a cloud service that helps developers easily build real-time features in web applications with publish-subscribe patterns at scale.
+
+Any scenario that requires real-time messaging between server and clients or among clients following publish-subscribe patterns can benefit from using Web PubSub. Developers no longer need to poll the server by sending repeated HTTP requests at intervals, which is wasteful and hard-to-scale.
+
+As shown in the diagram below, your clients establish WebSocket connections with your Web PubSub resource. This client library:
+
+- simplifies managing client connections
+- simplifies sending messages among clients
+- automatically retries after unintended drops of client connection
+- reliably deliveries messages in number and in order after recovering from connection drops
+
+![overflow](https://user-images.githubusercontent.com/668244/140014067-25a00959-04dc-47e8-ac25-6957bd0a71ce.png)
+
+Details about the terms used here are described in [key concepts](#key-concepts) section.
+
+_This library is hosted on [pypi][pypi]._
+
+## Getting started
+
+### Currently supported environments
+
+- [Python 3.7+](https://www.python.org/downloads/)
+
+### Prerequisites
+
+- An [Azure subscription][azure_sub].
+- A [Web PubSub resource][create_instance]
+
+### 1. Install the `azure-webpubsub-client` package
+
+```bash
+pip install azure-webpubsub-client
+```
+
+### 2. Connect with your Web PubSub resource
+
+A client uses a Client Access URL to connect and authenticate with the service, which follows a pattern of `wss://<service_name>.webpubsub.azure.com/client/hubs/<hub_name>?access_token=<token>`. A client can have a few ways to obtain the Client Access URL. For this quick start, you can copy and paste one from Azure Portal shown below.
+
+![get_client_url](https://learn.microsoft.com/azure/azure-web-pubsub/media/howto-websocket-connect/generate-client-url.png)
+
+As shown in the diagram above, the client has the permissions to send messages to and join a specific group named "_group1_". 
+
+```python
+from azure.webpubsub.client import WebPubSubClient
+
+client = WebPubSubClient("<<client-access-url>>")
+with client:
+    # The client can join/leave groups, send/receive messages to and from those groups all in real-time
+    ...
+```
+
+### 3. Join groups
+
+Note that a client can only receive messages from groups that it has joined and you need to add a callback to specify the logic when receiving messages.
+
+```python
+# ...continues the code snippet from above
+
+# Registers a listener for the event 'group-message' early before joining a group to not miss messages
+group_name = "group1";
+client.on("group-message", lambda e: print(f"Received message: {e.data}"));
```

Comment:
While it's not blocking for first preview - this still looks very JS. We need to do some Python ecosystem research on appropriate patterns here.

Reply:
Yeah. The only reason to use lambda here is to make the README cleaner. In [sample](https://github.com/Azure/azure-sdk-for-python/blob/webpubsub-client/sdk/webpubsub/azure-webpubsub-client/samples/hello_world.py) code, we use callback function.

Reply:
My comment was not in regards to the use of `lambda`, but rather having a method called `on` used to register handlers. However the use of `lamda` is also a bit problematic because there is no way to remove the handler (i.e. we have nothing to pass into the `off` method.